tests: drivers: build_all: Remove dts_fixup.h replace with overlay

galak opened the pull request:
Remove the dts_fixup.h with a DTS app overlay file instead.  We have
    sensors that show up on both the I2C and SPI busses, and we comment out
    the sensor to match how the tests work currently.
    
    The app.overlay utilizes the testing dts bindings to create test nodes
    that should not conflict with any existing SoC nodes.  Thus allowing the
    test to run on any platform.

galak commented:
Posting this in prep of #23245 being merged (as its based on it to get the test,* dts bindings).

mbolivar-nordic commented on tests/drivers/build_all/app.overlay:
```diff
@@ -0,0 +1,1029 @@
+			};
+
+			/*
+			lis2ds12@37 {
```
I read the description saying these are commented out to match how the test works today, but I don't get why that is necessary with the move to DT. Disabled nodes shouldn't have any effect, right? What am I missing?

galak replied:
Nothing, wasn't thinking to just set 'status = disabled'.  Will make that change as I like that solution better.

MaureenHelm commented on tests/drivers/build_all/app.overlay:
```diff
@@ -0,0 +1,1035 @@
+				/* read-only; */
+			};
+
+			mcr20a@10 {
```
Did you say you generated this overlay with a script? There seem to be additional nodes that weren't previously in the test. Can we keep the nodes in alpha order?

galak replied:
I had it general all I2C or SPI client devices, thus we may have more than was in dts_fixup.h.  While we can sort it alphabetically, the issue is that the reg value needs to be unique.  So lets say we have:

node_a / reg = 1
node_c / reg = 2

what happens when we add node_b, what reg value would it have?

The dts_fixup.h used the same reg value for everything, but when we build with DTS we'll get a DTS compile warning if we have the same reg value.

mbolivar-nordic replied:
Would it make sense to put each device node on its own vnd,i2c or vnd,spi bus? Then the device regs would be independent. The addresses of the buses don't matter as long as they're different.

bus-1 { node_a / reg = 1 }
bus-2 { node_c / reg = 1 }

---->

bus-1 { node_a / reg = 1 }
bus-2 { node_b / reg = 1 }
bus-3 { node_c / reg = 1 }

galak replied:
@mbolivar-nordic while we can do that, not sure what benefit that creates.  It means whenever someone adds a new device they're going to have to add a new controller which seems worse to me, than just having them add the device at the end of the list.

galak committed:
tests: drivers: build_all: Remove dts_fixup.h replace with overlay

Remove the dts_fixup.h with a DTS app overlay file instead.  We have
sensors that show up on both the I2C and SPI busses, and we disable
the sensors to match how the tests work currently.

The app.overlay utilizes the testing dts bindings to create test nodes
that should not conflict with any existing SoC nodes.  Thus allowing the
test to run on any platform.

The dts overlay is structured such that i2c.dtsi and spi.dtsi are
generated with default values for required properties.  We than update
device properties for those nodes that the defaults are sufficient for
one reason or another.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak commented:
The one CI failure is on mps2_an385 is known to fail `tests/drivers/counter/counter_basic_api/drivers.counter`.  Going to merge w/this known ci failure.
